src/lsstseries/ensemble.py

```diff
@@ -475,27 +482,52 @@ def _generate_object_table(self):
             .pivot_table(values=self._time_col, index=self._id_col, columns=self._band_col, aggfunc="sum")
         )
 
+        if keep_zero_entries:
+            # Determine which object IDs are missing from the source table.
+            object_inds = self._object.index.values.compute()
+            source_inds = self._source.index.values.compute()
+            missing_inds = np.setdiff1d(object_inds, source_inds).tolist()
+
+            # Create a dataframe of the missing IDs with zeros for all bands and counts.
+            rows = {self._id_col: missing_inds}
+            for i in res.columns.values:
+                rows[i] = [0] * len(missing_inds)
+
+            zero_pdf = pd.DataFrame(rows, dtype=int).set_index(self._id_col)
+            zero_ddf = dd.from_pandas(zero_pdf, sort=True, npartitions=1)
+
+            # Concatonate the zero dataframe onto the results.
+            res = dd.concat([res, zero_ddf], interleave_partitions=True).astype(int)
+
         # Rename bands to nobs_[band]
         band_cols = {col: f"nobs_{col}" for col in list(res.columns)}
         res = res.rename(columns=band_cols)
 
-        # Add total nobs
-        res[self._nobs_col] = counts.groupby(self._id_col).agg("sum")
+        # Add total nobs by summing across each band.
+        res[self._nobs_col] = res.sum(axis=1)
 
         return res
 
-    def _sync_tables(self):
-        """Sync operation to align both tables"""
+    def _sync_tables(self, keep_zero_entries=False):
```

Comment:
With this as a kwarg for a hidden function, should we add an ensemble flag to control this? Ensemble.keep_empty_objects or something? If we do, small additional note that the first generation of table from source in read_parquet will always want keep_zero_entries set to False, regardless of the flag, as we wouldn't have an existing ensemble._object to compare to.

Reply:
That's a really good call. I was wondering what the best way to handle turning it off and on (and clearly decided to hold off making that decision). I'm happy to change the code to enable by flag though.

Reply:
I've updated the code.

Reply:
Nice, looks good!